nav2_navfn_planner/src/navfn_planner.cpp

```diff
@@ -246,19 +246,20 @@ NavfnPlanner::smoothApproachToGoal(
 {
   // Replace the last pose of the computed path if it's actually further away
   // to the second to last pose than the goal pose.
-
-  auto second_to_last_pose = plan.poses.end()[-2];
-  auto last_pose = plan.poses.back();
-  if (
-    squared_distance(last_pose.pose, second_to_last_pose.pose) >
-    squared_distance(goal, second_to_last_pose.pose))
-  {
-    plan.poses.back().pose = goal;
-  } else {
-    geometry_msgs::msg::PoseStamped goal_copy;
-    goal_copy.pose = goal;
-    plan.poses.push_back(goal_copy);
+  if (plan.poses.size() > 2){
```

Comment:
```suggestion
  if (plan.poses.size() > 2) {
```

Reply:
Corrected the changes requested. Regarding retarget to master, does that mean the eloquent- devel will not have these changes? or master will be merged later to eloquent-devel

Reply:
Also, should I create a new pull request for the master branch with cherrypicked commits from the current pull request?

Reply:
Master may be eventually synced with eloquent, but all new development goes to master. You may create a new PR or rebase.

Reply:
https://github.com/ros-planning/navigation2/pull/1612 have raised the new PR, we can close this PR